src/transforms/gate/config.rs

```diff
@@ -0,0 +1,96 @@
+use vector_lib::config::{clone_input_definitions, LogNamespace};
+use vector_lib::configurable::configurable_component;
+
+use crate::{
+    conditions::AnyCondition,
+    config::{
+        DataType, GenerateConfig, Input, OutputId, TransformConfig, TransformContext,
+        TransformOutput,
+    },
+    schema,
+    transforms::Transform,
+};
+
+use super::transform::Gate;
+
+
+/// Configuration for the `gate` transform.
+#[configurable_component(transform(
+"gate",
+"Open or close an event stream based on supplied criteria"
+))]
+#[derive(Clone, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct GateConfig {
+    /// A logical condition used to pass events through without gating.
+    pub pass_when: Option<AnyCondition>,
+
+    /// A logical condition used to open the gate.
+    pub open_when: Option<AnyCondition>,
```

Comment:
Do we want to require this option? It's unclear to me what the behavior should be if there is no `open_when` 🤔